### PR TITLE
feat(resource): implement service detail in new architecture

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

Refacto of Resource Status service detail

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [x] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

Jira details

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by a-launois — APPROVED** [review]

File: src/Core/Application/RealTime/Common/RealTimeResponseTrait.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+trait RealTimeResponseTrait
+{
+    /**
+     * Converts an Icon model into an array
+     *
+     * @param Icon|null $icon
+     * @return array<string, string|null>
+     */
+    public static function iconToArray(?Icon $icon): array
```
Comment (kduret) [inline src/Core/Application/RealTime/Common/RealTimeResponseTrait.php:38]:
```suggestion
    public function iconToArray(?Icon $icon): array
```

File: src/Core/Application/RealTime/Common/RealTimeResponseTrait.php [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            ];
+    }
+
+    /**
+     * Converts an array of Downtimes entities into an array
+     *
+     * @param Downtime[] $downtimes
+     * @return array<int, array<string, mixed>>
+     */
+    public static function downtimesToArray(array $downtimes): array
```
Comment (kduret) [inline src/Core/Application/RealTime/Common/RealTimeResponseTrait.php:54]:
```suggestion
    public function downtimesToArray(array $downtimes): array
```

File: src/Core/Application/RealTime/Common/RealTimeResponseTrait.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        );
+    }
+
+    /**
+     * Converts an Acknowledgement entity into an array
+     *
+     * @param Acknowledgement|null $acknowledgement
+     * @return array<string, mixed>
+     */
+    public static function acknowledgementToArray(?Acknowledgement $acknowledgement): array
```
Comment (kduret) [inline src/Core/Application/RealTime/Common/RealTimeResponseTrait.php:86]:
```suggestion
    public function acknowledgementToArray(?Acknowledgement $acknowledgement): array
```

File: src/Core/Application/RealTime/Common/RealTimeResponseTrait.php [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+            ];
+    }
+
+    /**
+     * Converts Status model into an array for DTO
+     *
+     * @param Status $status
+     * @return array<string, mixed>
+     */
+    private static function statusToArray(Status $status): array
```
Comment (kduret) [inline src/Core/Application/RealTime/Common/RealTimeResponseTrait.php:115]:
```suggestion
    private function statusToArray(Status $status): array
```

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php [hunk 5]
```diff
@@ -73,6 +78,106 @@ public function __invoke(
...
+                $this->critical(
+                    "Host not found",
+                    [
+                        'id' => $hostId,
+                        'userId' => $this->contact->getId()
+                    ]
+                );
+                $presenter->setResponseStatus(new NotFoundResponse('Host'));
+                return;
+            }
```
Comment (kduret) [inline src/Core/Application/RealTime/UseCase/FindService/FindService.php:102]: create a private method

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php [hunk 6]
```diff
@@ -73,6 +78,106 @@ public function __invoke(
...
+                    "Service not found",
+                    [
+                        'id' => $serviceId,
+                        'hostId' => $hostId,
+                        'userId' => $this->contact->getId()
+                    ]
+                );
+                $presenter->setResponseStatus(new NotFoundResponse('Service'));
+                return;
+            }
```
Comment (kduret) [inline src/Core/Application/RealTime/UseCase/FindService/FindService.php:115]: create a private method

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php [hunk 7]
```diff
@@ -73,6 +78,106 @@ public function __invoke(
...
+                $this->critical(
+                    "Host not found",
+                    [
+                        'id' => $hostId,
+                        'userId' => $this->contact->getId()
+                    ]
+                );
+                $presenter->setResponseStatus(new NotFoundResponse('Host'));
+                return;
+            }
```
Comment (kduret) [inline src/Core/Application/RealTime/UseCase/FindService/FindService.php:140]: use private method

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php [hunk 8]
```diff
@@ -73,6 +78,106 @@ public function __invoke(
...
+                    "Service not found",
+                    [
+                        'id' => $serviceId,
+                        'hostId' => $hostId,
+                        'userId' => $this->contact->getId()
+                    ]
+                );
+                $presenter->setResponseStatus(new NotFoundResponse('Service'));
+                return;
+            }
```
Comment (kduret) [inline src/Core/Application/RealTime/UseCase/FindService/FindService.php:155]: use private method

File: src/Core/Application/RealTime/Common/RealTimeResponseTrait.php [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+    private function statusToArray(Status $status): array
+    {
+        return [
+            'name' => $status->getName(),
+            'code' => $status->getCode(),
+            'severity_code' => $status->getOrder(),
+            'type' => $status->getType()
+        ];
+    }
+}
```
Comment (callapa) [inline src/Core/Application/RealTime/Common/RealTimeResponseTrait.php:124]: You forgot to add a blank line at the end of the file.

File: src/Core/Application/RealTime/UseCase/FindHost/FindHost.php [hunk 10]
```diff
@@ -80,14 +81,14 @@ public function __invoke(int $hostId, FindHostPresenterInterface $presenter): vo
         if ($this->contact->isAdmin()) {
             $host = $this->repository->findHostById($hostId);
             if ($host === null) {
-                $this->debug(
+                $this->critical(
```
Comment (callapa) [inline src/Core/Application/RealTime/UseCase/FindHost/FindHost.php:84]:
```suggestion
                $this->error(
```

File: src/Core/Application/RealTime/UseCase/FindHost/FindHost.php [hunk 11]
```diff
@@ -99,14 +100,14 @@ public function __invoke(int $hostId, FindHostPresenterInterface $presenter): vo
             );
             $host = $this->repository->findHostByIdAndAccessGroupIds($hostId, $accessGroupIds);
             if ($host === null) {
-                $this->debug(
+                $this->critical(
```
Comment (callapa) [inline src/Core/Application/RealTime/UseCase/FindHost/FindHost.php:103]:
```suggestion
                $this->error(
```

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php [hunk 12]
```diff
@@ -73,6 +78,106 @@ public function __invoke(
...
+            "Searching details for service",
+            [
+                "id" => $serviceId
+            ]
+        );
+
+        if ($this->contact->isAdmin()) {
+            $host = $this->hostRepository->findHostById($hostId);
+            if ($host === null) {
+                $this->critical(
```
Comment (callapa) [inline src/Core/Application/RealTime/UseCase/FindService/FindService.php:93]:
```suggestion
                $this->error(
```

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php [hunk 13]
```diff
@@ -73,6 +78,106 @@ public function __invoke(
...
+                        'id' => $hostId,
+                        'userId' => $this->contact->getId()
+                    ]
+                );
+                $presenter->setResponseStatus(new NotFoundResponse('Host'));
+                return;
+            }
+            $service = $this->repository->findServiceById($hostId, $serviceId);
+            if ($service === null) {
+                $this->critical(
```
Comment (callapa) [inline src/Core/Application/RealTime/UseCase/FindService/FindService.php:105]:
```suggestion
                $this->error(
```

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php [hunk 14]
```diff
@@ -73,6 +78,106 @@ public function __invoke(
...
+            $accessGroups = $this->accessGroupRepository->findByContact($this->contact);
+            $accessGroupIds = array_map(
+                fn (AccessGroup $accessGroup) => $accessGroup->getId(),
+                $accessGroups
+            );
+
+            $host = $this->hostRepository->findHostByIdAndAccessGroupIds($hostId, $accessGroupIds);
+
+            if ($host === null) {
+                $this->critical(
```
Comment (callapa) [inline src/Core/Application/RealTime/UseCase/FindService/FindService.php:131]:
```suggestion
                $this->error(
```

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php [hunk 15]
```diff
@@ -73,6 +78,106 @@ public function __invoke(
...
+                    ]
+                );
+                $presenter->setResponseStatus(new NotFoundResponse('Host'));
+                return;
+            }
+
+            $service = $this->repository->findServiceByIdAndAccessGroupIds($hostId, $serviceId, $accessGroupIds);
+
+            if ($service === null) {
+                $this->critical(
```
Comment (callapa) [inline src/Core/Application/RealTime/UseCase/FindService/FindService.php:145]:
```suggestion
                $this->error(
```

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php [hunk 16]
```diff
@@ -84,8 +189,86 @@ public function __invoke(
...
+        ) {
+            try {
+                $legacyService = (new LegacyService())
+                    ->setId($service->getId())
+                    ->setCheckCommand($service->getCommandLine());
+
+                $this->monitoringService->hidePasswordInServiceCommandLine($legacyService);
+                $obfuscatedCommandLine = $legacyService->getCheckCommand();
+            } catch (\Throwable $ex) {
+                $this->debug(
```
Comment (callapa) [inline src/Core/Application/RealTime/UseCase/FindService/FindService.php:258]:
```suggestion
                $this->error(
```

**Review by callapa — CHANGES_REQUESTED** [review]

File: src/Core/Application/RealTime/UseCase/FindHost/FindHost.php [hunk 17]
```diff
@@ -80,14 +80,8 @@ public function __invoke(int $hostId, FindHostPresenterInterface $presenter): vo
...
-                $this->debug(
-                    "Host not found",
-                    [
-                        'id' => $hostId,
-                        'userId' => $this->contact->getId()
-                    ]
-                );
-                $presenter->setResponseStatus(new HostNotFoundResponse());
+                $this->handleHostNotFound($hostId, $presenter);
+                $presenter->setResponseStatus(new NotFoundResponse('Host'));
```
Comment (kduret) [inline src/Core/Application/RealTime/UseCase/FindHost/FindHost.php:84]:
```suggestion
```

**Review by kduret — APPROVED** [review]

File: src/Core/Application/RealTime/UseCase/FindHost/FindHost.php [hunk 18]
```diff
@@ -81,13 +80,7 @@ public function __invoke(int $hostId, FindHostPresenterInterface $presenter): vo
...
             if ($host === null) {
-                $this->critical(
-                    "Host not found",
-                    [
-                        'id' => $hostId,
-                        'userId' => $this->contact->getId()
-                    ]
-                );
+                $this->handleHostNotFound($hostId, $presenter);
                 $presenter->setResponseStatus(new NotFoundResponse('Host'));
```
Comment (callapa) [inline src/Core/Application/RealTime/UseCase/FindHost/FindHost.php]: You forgot to delete this line.

**Review by callapa — CHANGES_REQUESTED** [review]

File: src/Core/Application/RealTime/UseCase/FindHost/FindHost.php (hunk 18 above)
Comment (adr-mo) [reply src/Core/Application/RealTime/UseCase/FindHost/FindHost.php]: Already patched here https://github.com/centreon/centreon/pull/10589/commits/5565a27029cf25fee3b353e442d98acbaebf020f

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php (hunk 16 above)
Comment (adr-mo) [reply src/Core/Application/RealTime/UseCase/FindService/FindService.php:268]: ✅  done

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php (hunk 14 above)
Comment (adr-mo) [reply src/Core/Application/RealTime/UseCase/FindService/FindService.php]: ✅  done

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php (hunk 13 above)
Comment (adr-mo) [reply src/Core/Application/RealTime/UseCase/FindService/FindService.php]: ✅  done

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php (hunk 12 above)
Comment (adr-mo) [reply src/Core/Application/RealTime/UseCase/FindService/FindService.php]: ✅  done

File: src/Core/Application/RealTime/UseCase/FindHost/FindHost.php (hunk 11 above)
Comment (adr-mo) [reply src/Core/Application/RealTime/UseCase/FindHost/FindHost.php]: ✅  done

File: src/Core/Application/RealTime/UseCase/FindHost/FindHost.php (hunk 10 above)
Comment (adr-mo) [reply src/Core/Application/RealTime/UseCase/FindHost/FindHost.php]: ✅  done

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php (hunk 7 above)
Comment (adr-mo) [reply src/Core/Application/RealTime/UseCase/FindService/FindService.php]: ✅  done

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php (hunk 6 above)
Comment (adr-mo) [reply src/Core/Application/RealTime/UseCase/FindService/FindService.php]: ✅  done

File: src/Core/Application/RealTime/UseCase/FindService/FindService.php (hunk 5 above)
Comment (adr-mo) [reply src/Core/Application/RealTime/UseCase/FindService/FindService.php]: ✅  done

File: src/Core/Application/RealTime/UseCase/FindHost/FindHost.php (hunk 18 above)
Comment (adr-mo) [reply src/Core/Application/RealTime/UseCase/FindHost/FindHost.php]: ✅  done

**Review by callapa — APPROVED** [review]

**Pull request merged**